# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/*/metamail/20_mm-2.7-uudecode.patch
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- mm2.7/src/bin/sun-audio-file.ewt	Fri Oct 24 09:41:11 1997
+++ mm2.7/src/bin/sun-audio-file	Fri Oct 24 09:52:05 1997
@@ -6,7 +6,13 @@
     set METAMAIL_TMPDIR=/tmp
 endif
 
-cd ${METAMAIL_TMPDIR}
-uudecode < $1
-audiotool audio-file
-rm -f audio-file $1
+set fn=`mktemp $METAMAIL_TMPDIR/audio.XXXXXX`
+test -n "$fn"
+if $? then
+	exit 1
+endif
+
+uudecode -o $fn < $1
+
+audiotool $fn
+rm -f $fn $1
--- mm2.7/src/bin/sun-message.csh.ewt	Fri Oct 24 09:44:07 1997
+++ mm2.7/src/bin/sun-message.csh	Fri Oct 24 09:52:30 1997
@@ -9,7 +9,13 @@
 cd /tmp/decode.$$
 
 if ($2 == "uuencode") then
-    uudecode $1
+    set fn=`mktemp $METAMAIL_TMPDIR/audio.XXXXXX`
+    test -n "$fn"
+    if $? then
+	    exit 1
+    endif
+
+    uudecode -o $fn $1
 
     echo "The following file was uudecoded:"
     echo ""
@@ -17,7 +23,6 @@
     set defans = "1"
     while (1)
 	ls -l
-	set fn = *
 
 	echo ""
 	echo "Please choose one:"
